src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/close_application_request.cc

```diff
@@ -0,0 +1,157 @@
+ POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "sdl_rpc_plugin/commands/mobile/close_application_request.h"
+#include "application_manager/rpc_service.h"
+#include "application_manager/message_helper.h"
+#include "utils/include/utils/helpers.h"
+
+namespace sdl_rpc_plugin {
+using namespace application_manager;
+
+namespace commands {
+
+CloseApplicationRequest::CloseApplicationRequest(
+    const app_mngr::commands::MessageSharedPtr& message,
+    app_mngr::ApplicationManager& application_manager,
+    app_mngr::rpc_service::RPCService& rpc_service,
+    app_mngr::HMICapabilities& hmi_capabilities,
+    policy::PolicyHandlerInterface& policy_handler)
+    : CommandRequestImpl(message,
+                         application_manager,
+                         rpc_service,
+                         hmi_capabilities,
+                         policy_handler) {}
+
+CloseApplicationRequest::~CloseApplicationRequest() {}
+
+void CloseApplicationRequest::Run() {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  ApplicationSharedPtr application =
+      application_manager_.application(connection_key());
+
+  if (!application) {
+    LOG4CXX_ERROR(logger_, "Application is not registered");
+    SendResponse(false, mobile_apis::Result::APPLICATION_NOT_REGISTERED);
+    return;
+  }
+
+  const mobile_apis::HMILevel::eType app_hmi_level = application->hmi_level();
+
+  if (mobile_apis::HMILevel::HMI_NONE == app_hmi_level) {
+    LOG4CXX_DEBUG(logger_,
+                  "Application with id " << connection_key()
+                                         << " is already in hmi_level NONE.");
+    SendResponse(false, mobile_apis::Result::IGNORED);
+  }
+
+  uint32_t activate_app_corr_id = 0;
+  const bool sent_success = SendBCActivateApp(
+      application, hmi_apis::Common_HMILevel::NONE, true, activate_app_corr_id);
+
+  if (sent_success) {
+    subscribe_on_event(hmi_apis::FunctionID::BasicCommunication_ActivateApp,
+                       activate_app_corr_id);
+  } else {
+    LOG4CXX_ERROR(logger_, "Unable to send BC.ActivateApp");
+    SendResponse(false, mobile_apis::Result::IGNORED);
+  }
+}
+
+void CloseApplicationRequest::on_event(const event_engine::Event& event) {
+  using namespace helpers;
+  LOG4CXX_AUTO_TRACE(logger_);
+  const smart_objects::SmartObject& message = event.smart_object();
+
+  if (hmi_apis::FunctionID::BasicCommunication_ActivateApp != event.id()) {
+    LOG4CXX_ERROR(logger_, "Received unknown event" << event.id());
+    return;
+  }
+
+  const auto hmi_result = static_cast<hmi_apis::Common_Result::eType>(
+      message[strings::params][hmi_response::code].asInt());
+
+  const auto mobile_result = MessageHelper::HMIToMobileResult(hmi_result);
+  const bool success = Compare<hmi_apis::Common_Result::eType, EQ, ONE>(
+      hmi_result,
+      hmi_apis::Common_Result::SUCCESS,
+      hmi_apis::Common_Result::WARNINGS);
+
+  auto application = application_manager_.application(connection_key());
+  if (success) {
```

Comment:
@IGapchuk I would also check application pointer.
And if success != true, you do not need to look for application

Reply:
@LuxoftAKutsan   Fixed in 4257effee21e05a960a02a4b14365091e16200b7.